test-data/unit/plugins/attrhook.py

```diff
@@ -0,0 +1,21 @@
+from typing import Optional, Callable
+
+from mypy.plugin import Plugin, AttributeContext
+from mypy.types import Type, Instance
+
+
+class AttrPlugin(Plugin):
+    def get_attribute_hook(self, fullname: str) -> Optional[Callable[[AttributeContext], Type]]:
+        if fullname == 'm.Signal.__call__':
+            return signal_call_callback
+        return None
+
+
+def signal_call_callback(ctx: AttributeContext) -> Type:
+    if isinstance(ctx.type, Instance) and ctx.type.type.fullname() == 'm.Signal':
+        return ctx.type.args[0]
+    return ctx.inferred_attr_type
```

Comment:
Would there be value in changing the protocol so that if a callback/hook returns None this means it didn't come up with anything and the plugin framework will return the originally inferred type?